modules/plottinggl/src/plotters/scatterplotgl.cpp

```diff
@@ -474,86 +241,98 @@ void ScatterPlotGL::setYAxis(const Column* col) {
 
 void ScatterPlotGL::setXAxisData(const Column* col) {
     if (col) {
-        xAxis_ = col->getBuffer();
+        points_.xCoord = col->getBuffer();
         minmaxX_ = vec2(col->getRange());
         properties_.xAxis_.setRange(minmaxX_);
     } else {
-        xAxis_ = nullptr;
+        points_.xCoord = nullptr;
     }
-    boxSelectionHandler_.setXAxisData(xAxis_);
+    partitionDirty_ = true;
```

Comment:
partition?

Reply:
Same idea as in the PCP. Consider the indices in one container. This container is partitioned/divided into multiple blocks (filtered, normal, selected, and highlighted).